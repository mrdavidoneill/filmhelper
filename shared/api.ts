import { FilmInfoType, FilmSearchResponseType } from "@/shared/types";
const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL;

export async function postAPIRequest({
  body,
  resource,
}: {
  body: Object;
  resource: string;
}) {
  const response = await fetch(`${API_BASE_URL}/${resource}/`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(body),
  });
  return response.json();
}

export async function getAPIRequest({
  token,
  resource,
}: {
  token: string;
  resource: string;
}) {
  const response = await fetch(`${API_BASE_URL}/${resource}/`, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Token ${token}`,
    },
  });
  return response.json();
}

export async function fetchToken(username: string, password: string) {
  return postAPIRequest({
    body: { username, password },
    resource: "api-token-auth",
  });
}

export async function getFilmWatchList(token: string) {
  return getAPIRequest({
    token,
    resource: "filmwatchlist",
  });
}

export async function searchFilms(
  input: string
): Promise<FilmSearchResponseType> {
  const response = await fetch(
    `http://www.omdbapi.com/?s=${input}&type=movie&apikey=${process.env.NEXT_PUBLIC_OMDB_KEY}`
  );
  return response.json();
}

export async function getFilmInfo(imdbID: string): Promise<FilmInfoType> {
  const response = await fetch(
    `http://www.omdbapi.com/?i=${imdbID}&apikey=${process.env.NEXT_PUBLIC_OMDB_KEY}`
  );
  return response.json();
}
