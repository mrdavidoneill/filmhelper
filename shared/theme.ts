import { blue, deepPurple } from "@mui/material/colors";
import { createTheme } from "@mui/material/styles";
export const theme = createTheme({
  palette: {
    primary: blue,
    secondary: deepPurple,
  },
});
