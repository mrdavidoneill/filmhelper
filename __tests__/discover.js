const GOOD_SEARCH_FILMS = require("@/shared/api_examples/movie_search.json");
const BAD_SEARCH_FILMS = require("@/shared/api_examples/movie_search_error.json");
const GOOD_FILM_INFO = require("@/shared/api_examples/movie_info.json");

import "@testing-library/jest-dom";

import React from "react";
import { render, act, fireEvent, waitFor } from "@testing-library/react";
import { getFilmInfo, searchFilms } from "@/shared/api";
import DiscoverPage from "@/pages/discover";

jest.mock("next/router", () => ({
  useRouter: jest.fn().mockReturnValue({
    push: jest.fn(),
  }),
}));

jest.mock("@/shared/api", () => {
  return {
    searchFilms: jest.fn((input) => {
      if (input.length > 2) {
        return Promise.resolve(GOOD_SEARCH_FILMS);
      } else {
        return Promise.resolve(BAD_SEARCH_FILMS);
      }
    }),
    getFilmInfo: jest.fn(() => {
      return Promise.resolve(GOOD_FILM_INFO);
    }),
  };
});
const localStorageMock = {
  getItem: jest.fn().mockReturnValue("token"),
};

describe("DiscoverPage", () => {
  it("renders without error", () => {
    global.localStorage = localStorageMock;
    const { asFragment } = render(<DiscoverPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("handles search and sets items", async () => {
    global.localStorage = localStorageMock;
    const { getByPlaceholderText, getByText } = render(<DiscoverPage />);

    const input = getByPlaceholderText("Search…");
    act(() => {
      fireEvent.change(input, { target: { value: "Rocky" } });
    });
    act(() => {
      fireEvent.keyDown(input, { key: "Enter", code: 13 });
    });
    await waitFor(() => {
      expect(searchFilms).toHaveBeenCalledWith("Rocky");
    });
    await waitFor(() => {
      expect(getFilmInfo).toHaveBeenCalledWith(
        GOOD_SEARCH_FILMS.Search[0].imdbID
      );
    });
  });

  it("handles no search results", async () => {
    global.localStorage = localStorageMock;
    const { getByPlaceholderText, getByText } = render(<DiscoverPage />);

    const input = getByPlaceholderText("Search…");
    act(() => {
      fireEvent.change(input, { target: { value: "A" } });
    });
    act(() => {
      fireEvent.keyDown(input, { key: "Enter", code: 13 });
    });
    await waitFor(() => {
      expect(searchFilms).toHaveBeenCalledWith("A");
    });

    await waitFor(() => {
      expect(getByText(/Error/)).toBeInTheDocument();
    });
  });
});
