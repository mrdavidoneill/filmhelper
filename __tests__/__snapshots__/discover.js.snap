// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscoverPage renders without error 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col bg-slate-300 h-[100vh] overflow-hidden animate-fadein"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic animate-goright css-4issou-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-dudo9s-MuiTypography-root"
        >
          Film buddies
        </div>
        <div
          class="css-1orzy9s"
        >
          <div
            class="css-1iz4iln"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-7mt4h1-MuiInputBase-root"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              placeholder="Searchâ€¦"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </header>
    <div
      class="w-full animate-fadeinslow"
    >
      <div
        class="bg-black cursor-pointer outline-white flickity-enabled is-draggable"
        tabindex="0"
      >
        <div
          class="flickity-viewport"
          style="height: 0px;"
        >
          <div
            class="flickity-slider"
            style="left: 0px; transform: translateX(NaN%);"
          >
            <div
              class="flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center is-selected"
              data-testid="https://m.media-amazon.com/images/M/MV5BMjAxMzY3NjcxNF5BMl5BanBnXkFtZTcwNTI5OTM0Mw@@._V1_SX300.jpg"
              style="background-image: url(https://m.media-amazon.com/images/M/MV5BMjAxMzY3NjcxNF5BMl5BanBnXkFtZTcwNTI5OTM0Mw@@._V1_SX300.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
            />
            <div
              aria-hidden="true"
              class=" flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center"
              data-testid="https://m.media-amazon.com/images/M/MV5BYzE5MjY1ZDgtMTkyNC00MTMyLThhMjAtZGI5OTE1NzFlZGJjXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SX300.jpg"
              style="background-image: url(https://m.media-amazon.com/images/M/MV5BYzE5MjY1ZDgtMTkyNC00MTMyLThhMjAtZGI5OTE1NzFlZGJjXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SX300.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
            />
            <div
              aria-hidden="true"
              class=" flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center"
              data-testid="https://m.media-amazon.com/images/M/MV5BNDYxNjQyMjAtNTdiOS00NGYwLWFmNTAtNThmYjU5ZGI2YTI1XkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SX300.jpg"
              style="background-image: url(https://m.media-amazon.com/images/M/MV5BNDYxNjQyMjAtNTdiOS00NGYwLWFmNTAtNThmYjU5ZGI2YTI1XkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SX300.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
            />
            <div
              aria-hidden="true"
              class=" flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center"
              data-testid="https://m.media-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SX300.jpg"
              style="background-image: url(https://m.media-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SX300.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
            />
            <div
              aria-hidden="true"
              class=" flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center"
              data-testid="https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SX300.jpg"
              style="background-image: url(https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SX300.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
            />
            <div
              aria-hidden="true"
              class=" flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center"
              data-testid="https://m.media-amazon.com/images/M/MV5BMTYwOTEwNjAzMl5BMl5BanBnXkFtZTcwODc5MTUwMw@@._V1_SX300.jpg"
              style="background-image: url(https://m.media-amazon.com/images/M/MV5BMTYwOTEwNjAzMl5BMl5BanBnXkFtZTcwODc5MTUwMw@@._V1_SX300.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
            />
            <div
              aria-hidden="true"
              class=" flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center"
              data-testid="https://m.media-amazon.com/images/M/MV5BMTAwMjU5OTgxNjZeQTJeQWpwZ15BbWU4MDUxNDYxODEx._V1_SX300.jpg"
              style="background-image: url(https://m.media-amazon.com/images/M/MV5BMTAwMjU5OTgxNjZeQTJeQWpwZ15BbWU4MDUxNDYxODEx._V1_SX300.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
            />
            <div
              aria-hidden="true"
              class=" flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center"
              data-testid="https://m.media-amazon.com/images/M/MV5BMmEzNTkxYjQtZTc0MC00YTVjLTg5ZTEtZWMwOWVlYzY0NWIwXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg"
              style="background-image: url(https://m.media-amazon.com/images/M/MV5BMmEzNTkxYjQtZTc0MC00YTVjLTg5ZTEtZWMwOWVlYzY0NWIwXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
            />
            <div
              aria-hidden="true"
              class=" flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center"
              data-testid="https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_SX300.jpg"
              style="background-image: url(https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_SX300.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
            />
            <div
              aria-hidden="true"
              class=" flex flex-grow mx-2 h-[35vh] w-[30vh] max-w-sm bg-cover bg-center"
              data-testid="https://m.media-amazon.com/images/M/MV5BMjIyNTQ5NjQ1OV5BMl5BanBnXkFtZTcwODg1MDU4OA@@._V1_SX300.jpg"
              style="background-image: url(https://m.media-amazon.com/images/M/MV5BMjIyNTQ5NjQ1OV5BMl5BanBnXkFtZTcwODg1MDU4OA@@._V1_SX300.jpg); position: absolute; left: 0px; transform: translateX(NaN%);"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col flex-1 overflow-y-auto animate-risefast"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root h-full self-center w-full max-w-3xl overflow-y-auto animate-fadein css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall self-center w-full css-1a0mp58-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Add to watchlist
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiCardHeader-root pb-0 css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Inception
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
            >
              Action, Adventure, Sci-Fi, Thriller
            </span>
          </div>
          <div
            class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
          >
            <div
              class="css-1okj3ks-MuiStack-root"
            >
              <span
                class="MuiRating-root MuiRating-sizeMedium css-iljtu2-MuiRating-root"
              >
                <span
                  class="MuiRating-decimal css-34he1w-MuiRating-decimal"
                >
                  <label
                    class="css-dqr9h-MuiRating-label"
                    for=":rc:"
                    style="width: 0%; overflow: hidden; position: absolute;"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      0.5 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":rc:"
                    name="half-rating"
                    type="radio"
                    value="0.5"
                  />
                  <label
                    class="css-dqr9h-MuiRating-label"
                    for=":rd:"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      1 Star
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":rd:"
                    name="half-rating"
                    type="radio"
                    value="1"
                  />
                </span>
                <span
                  class="MuiRating-decimal css-34he1w-MuiRating-decimal"
                >
                  <label
                    class="css-dqr9h-MuiRating-label"
                    for=":re:"
                    style="width: 0%; overflow: hidden; position: absolute;"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      1.5 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":re:"
                    name="half-rating"
                    type="radio"
                    value="1.5"
                  />
                  <label
                    class="css-dqr9h-MuiRating-label"
                    for=":rf:"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      2 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":rf:"
                    name="half-rating"
                    type="radio"
                    value="2"
                  />
                </span>
                <span
                  class="MuiRating-decimal css-34he1w-MuiRating-decimal"
                >
                  <label
                    class="css-dqr9h-MuiRating-label"
                    for=":rg:"
                    style="width: 0%; overflow: hidden; position: absolute;"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      2.5 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":rg:"
                    name="half-rating"
                    type="radio"
                    value="2.5"
                  />
                  <label
                    class="css-dqr9h-MuiRating-label"
                    for=":rh:"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      3 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":rh:"
                    name="half-rating"
                    type="radio"
                    value="3"
                  />
                </span>
                <span
                  class="MuiRating-decimal css-34he1w-MuiRating-decimal"
                >
                  <label
                    class="css-dqr9h-MuiRating-label"
                    for=":ri:"
                    style="width: 0%; overflow: hidden; position: absolute;"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      3.5 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":ri:"
                    name="half-rating"
                    type="radio"
                    value="3.5"
                  />
                  <label
                    class="css-dqr9h-MuiRating-label"
                    for=":rj:"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      4 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":rj:"
                    name="half-rating"
                    type="radio"
                    value="4"
                  />
                </span>
                <span
                  class="MuiRating-decimal css-34he1w-MuiRating-decimal"
                >
                  <label
                    class="css-dqr9h-MuiRating-label"
                    for=":rk:"
                    style="width: 0%; overflow: hidden; position: absolute;"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      4.5 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":rk:"
                    name="half-rating"
                    type="radio"
                    value="4.5"
                  />
                  <label
                    class="css-dqr9h-MuiRating-label"
                    for=":rl:"
                  >
                    <span
                      class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="StarBorderIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiRating-visuallyHidden"
                    >
                      5 Stars
                    </span>
                  </label>
                  <input
                    class="MuiRating-visuallyHidden"
                    id=":rl:"
                    name="half-rating"
                    type="radio"
                    value="5"
                  />
                </span>
                <label
                  class="MuiRating-label MuiRating-pristine css-dqr9h-MuiRating-label"
                >
                  <input
                    class="MuiRating-visuallyHidden"
                    id="half-rating-empty"
                    name="half-rating"
                    type="radio"
                    value=""
                  />
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    Empty
                  </span>
                </label>
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiCardContent-root pt-0 css-46bh2p-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            148 min | PG-13
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
          >
            Leonardo DiCaprio, Joseph Gordon-Levitt, Ellen Page, Tom Hardy
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Dom Cobb is a skilled thief, the absolute best in the dangerous art of extraction, stealing valuable secrets from deep within the subconscious during the dream state, when the mind is at its most vulnerable. Cobb's rare ability has made him a coveted player in this treacherous new world of corporate espionage, but it has also made him an international fugitive and cost him everything he has ever loved. Now Cobb is being offered a chance at redemption. One last job could give him his life back but only if he can accomplish the impossible - inception. Instead of the perfect heist, Cobb and his team of specialists have to pull off the reverse: their task is not to steal an idea but to plant one. If they succeed, it could be the perfect crime. But no amount of careful planning or expertise can prepare the team for the dangerous enemy that seems to predict their every move. An enemy that only Cobb could have seen coming.
          </p>
        </div>
      </div>
    </div>
    <div
      class="animate-rise MuiBox-root css-0"
    >
      <div
        class="MuiBottomNavigation-root w-full css-gumrph-MuiBottomNavigation-root"
      >
        <a
          class="MuiButtonBase-root MuiBottomNavigationAction-root css-11jxdpm-MuiButtonBase-root-MuiBottomNavigationAction-root"
          href="discover"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MovieIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m18 4 2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
            />
          </svg>
          <span
            class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
          >
            Discover
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButtonBase-root MuiBottomNavigationAction-root css-11jxdpm-MuiButtonBase-root-MuiBottomNavigationAction-root"
          href="watchlist"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
          <span
            class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
          >
            Watch List
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButtonBase-root MuiBottomNavigationAction-root css-11jxdpm-MuiButtonBase-root-MuiBottomNavigationAction-root"
          href="ratings"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RateReviewIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 14v-2.47l6.88-6.88c.2-.2.51-.2.71 0l1.77 1.77c.2.2.2.51 0 .71L8.47 14H6zm12 0h-7.5l2-2H18v2z"
            />
          </svg>
          <span
            class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
          >
            My Ratings
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
